src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
-                    }
-
-                    for (int col = 0; col < zoneSetInfo.columns(); col++, j += 2)
-                    {
-                        zoneSetInfo.columnsPercents()[col] = data[j] * 256 + data[j + 1];
-                    }
-
-                    for (int row = 0; row < zoneSetInfo.rows(); row++)
-                    {
-                        for (int col = 0; col < zoneSetInfo.columns(); col++)
-                        {
-                            zoneSetInfo.cellChildMap()[row][col] = data[j++];
-                        }
-                    }
-                    zoneSetData.info = zoneSetInfo;
-                    break;
-                }
-                case CustomLayoutType::Canvas:
-                {
-                    CanvasLayoutInfo info;
-
-                    int j = 5;
-                    info.lastWorkAreaWidth = data[j] * 256 + data[j + 1];
-                    j += 2;
-                    info.lastWorkAreaHeight = data[j] * 256 + data[j + 1];
-                    j += 2;
-
-                    int count = data[j++];
-                    info.zones.reserve(count);
-                    while (count-- > 0)
-                    {
-                        int x = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int y = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int width = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int height = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        info.zones.push_back(CanvasLayoutInfo::Rect{
-                            x, y, width, height });
-                    }
-                    zoneSetData.info = info;
-                    break;
-                }
-                default:
-                    continue;
-                }
-                customZoneSetsMap[uuid] = zoneSetData;
+            CustomZoneSetJSON result;
 
-                valueLength = ARRAYSIZE(value);
-                dataSize = ARRAYSIZE(data);
+            result.uuid = customZoneSet.GetNamedString(L"uuid");
+            if (!IsValidGuid(result.uuid))
+            {
+                return std::nullopt;
             }
-        }
-    }
 
-    void FancyZonesData::RemoveDesktopAppZoneHistory(const std::wstring& desktopId)
-    {
-        for (auto it = std::begin(appZoneHistoryMap); it != std::end(appZoneHistoryMap);)
-        {
-            auto& perDesktopData = it->second;
-            for (auto desktopIt = std::begin(perDesktopData); desktopIt != std::end(perDesktopData);)
+            result.data.name = customZoneSet.GetNamedString(L"name");
+
+            json::JsonObject infoJson = customZoneSet.GetNamedObject(L"info");
+            std::wstring zoneSetType = std::wstring{ customZoneSet.GetNamedString(L"type") };
+            if (zoneSetType.compare(L"canvas") == 0)
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.